experimental/dds/tree2/src/test/shared-tree/editing.spec.ts

```diff
@@ -1012,6 +1038,146 @@ describe("Editing", () => {
 
 				expectJsonTree([tree, tree2], [{ foo: "x" }]);
 			});
+
+			it("existence constraint on node inserted in prior transaction", () => {
+				const tree = makeTreeFromJson([]);
+				const tree2 = tree.fork();
+
+				// Insert "a"
+				// State should be: ["a"]
+				const sequence = tree.editor.sequenceField({
+					parent: undefined,
+					field: rootFieldKeySymbol,
+				});
+				sequence.insert(0, singleTextCursor({ type: jsonString.name, value: "a" }));
+
+				// Constrain on "a" and insert "b" into root sequence if it does
+				// State should be: ["a", "b"]
+				tree.transaction.start();
+				tree.editor.addNodeExistsConstraint({
+					parent: undefined,
+					parentField: rootFieldKeySymbol,
+					parentIndex: 0,
+				});
+				const rootSequence = tree.editor.sequenceField({
+					parent: undefined,
+					field: rootFieldKeySymbol,
+				});
+				rootSequence.insert(1, singleTextCursor({ type: jsonString.name, value: "b" }));
+				tree.transaction.commit();
+
+				// Make a concurrent edit to rebase over that inserts into root sequence
+				// State should be (to tree2): ["c"]
+				const tree2RootSequence = tree2.editor.sequenceField({
+					parent: undefined,
+					field: rootFieldKeySymbol,
+				});
+				tree2RootSequence.insert(
+					0,
+					singleTextCursor({ type: jsonString.name, value: "c" }),
+				);
+
+				tree.merge(tree2);
+				tree2.rebaseOnto(tree);
+
+				expectJsonTree([tree, tree2], ["c", "a", "b"]);
+			});
+
+			it("can add constraint to node inserted in same transaction", () => {
```

Comment:
We should have a variant of this test demonstrating that the transaction fails if the node with the constraint cannot be inserted.

Reply:
Added test for this scenario that is currently failing and skipped. Will follow with a PR that fixes it.